scipy/spatial/transform/_rotation.pyx

```diff
@@ -829,7 +829,10 @@ cdef class Rotation:
     @cython.wraparound(False)
     def __init__(self, quat, normalize=True, copy=True, scalar_first=False):
         self._single = False
-        quat = np.asarray(quat, dtype=float)
+        if normalize or copy:
```

Comment:
I don't think so? Since we need to cast single quats to `(1, 4)` arrays to perform the roll operation during the `scalar_first` check on line 850. And it looks like there isn't a way to really get around creating a new array with `np.roll`. https://stackoverflow.com/questions/35916201/alternative-to-numpy-roll-without-copying-array

Reply:
I don't immediately see that this change somehow reduces the number of copies.

If `np.asarray` doesn't do copies, then the initial code looks correct and optimal in this regard.

Reply:
Ah yeah I think you're right, reverted.